RecoMET/METProducers/src/CaloRecHitsBeamHaloCleaned.cc

```diff
@@ -0,0 +1,349 @@
+// -*- C++ -*-
+//
+// Package:    RecoMET/METProducers
+// Class:      CaloRecHitsBeamHaloCleaned
+// 
+/**\class CaloRecHitsBeamHaloCleaned CaloRecHitsBeamHaloCleaned.cc RecoMET/METProducers/plugins/CaloRecHitsBeamHaloCleaned.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Thomas Laurent
+//         Created:  Tue, 09 Feb 2016 13:09:37 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include <vector>
+#include <iostream>
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/METReco/interface/CSCHaloData.h"
+#include "RecoMET/METAlgorithms/interface/CSCHaloAlgo.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+
+
+class CaloRecHitsBeamHaloCleaned : public edm::stream::EDProducer<> {
+   public:
+      explicit CaloRecHitsBeamHaloCleaned(const edm::ParameterSet&);
+      ~CaloRecHitsBeamHaloCleaned();
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+  virtual void beginStream(edm::StreamID) override;
+      virtual void produce(edm::Event&, const edm::EventSetup&) override;
+      virtual void endStream() override;
+
+  edm::EDGetTokenT<reco::TrackCollection > trk_token; 
+  edm::EDGetTokenT<EcalRecHitCollection> ecalebhits_token;
+  edm::EDGetTokenT<EcalRecHitCollection> ecaleehits_token;
+  edm::EDGetTokenT<HBHERecHitCollection> hbhehits_token; 
+  edm::EDGetTokenT<reco::MuonCollection> cosmicmuon_token_;
+  edm::EDGetTokenT<reco::MuonTimeExtraMap> csctimemap_token_;
+  edm::EDGetTokenT<reco::MuonCollection> muon_token_;
+  edm::EDGetTokenT<CSCSegmentCollection> cscsegment_token_;
+  edm::EDGetTokenT<CSCRecHit2DCollection> cscrechit_token_;
+  edm::EDGetTokenT<CSCALCTDigiCollection> cscalct_token_;
+  edm::EDGetTokenT<L1MuGMTReadoutCollection> l1mugmtro_token_;
+  edm::EDGetTokenT<edm::TriggerResults> hltresult_token_;
+
+  //Input tags
+  edm::InputTag IT_L1MuGMTReadout;
+  edm::InputTag IT_ALCT;
+  edm::InputTag IT_HLTResult;
+  std::vector< edm::InputTag > vIT_HLTBit  ;
+  edm::InputTag IT_CSCRecHit;
+  edm::InputTag IT_CosmicMuon;
+  edm::InputTag IT_CSCSegment;
+  edm::InputTag IT_Muon;
+  edm::InputTag IT_EBRecHits;
+  edm::InputTag IT_EERecHits;
+  edm::InputTag IT_HBHERecHits;
+
+
+  CSCHaloAlgo CSCAlgo;
+  MuonSegmentMatcher * TheMatcher;
```

Comment:
CMSSW style is to start data members names with lower case.
Presence of "the" "m_" or trailing underscore is encouraged to differentiate data members from plain local variables in method implementations.